### DISCUSSION TITLE
$args as an object

### DISCUSSION THREAD

**Discussion by Afrowson** [nuwave/lighthouse#2615] [Ideas]
I was working on making a previously required ID optional for a specific mutation. When i adjusted everything, i wrote a test with NULL as ID but it obviously failed when the frontend omited the ID while sending it. Lighthouse andled everything well, but in my code it did break because i cant array access a key that does not exist... 

If the $args passed to the resolver would be an object like a Request or Collection, this could be handled gracefully by simply returning null for a missing key... If the Interrest for such a solution exists, i would be willing to invest some time in to it.

**Comment by spawnia** [comment]
I use [PHPStan array shapes](https://phpstan.org/writing-php-code/phpdoc-types#array-shapes) to type `$args`, this allows static analysis to ensure accesses are safe.

```php
/** @param array{id?: ?string} */
public function __invoke(null $_, array $args): ...
```

Note the two `?`, optional arguments in GraphQL can be null or omitted.

Also, have you considered `$args['id'] ?? null`?
